packages/aws-cdk-lib/aws-autoscaling/lib/step-scaling-policy.ts

```diff
@@ -102,6 +118,10 @@ export class StepScalingPolicy extends Construct {
       throw new Error(`'scalingSteps' can have at most 40 steps, got ${props.scalingSteps.length}`);
     }
 
+    if (props.datapointsToAlarm !== undefined && !Token.isUnresolved(props.datapointsToAlarm) && props.datapointsToAlarm < 1) {
```

Comment:
Can we set `datapointsToAlarm` to a value greater than `evaluationPeriods`?
If not, I think we should validate that case as well (and update the documentation for the property).

Reply:
CFn returns an error `DatapointsToAlarm must be less than or equal to EvaluationPeriods`. So I will add the validation and the documentation the property.